src/test/java/org/apache/commons/mail/resolver/DataSourceUrlResolverTest.java

```diff
@@ -81,13 +83,14 @@ public void testResolvingHttpLenientHost() throws Exception
      *
      * @throws Exception the test failed
      */
-    @Test(expected = IOException.class)
+    @Test
     public void testResolvingHttpNonLenient() throws Exception
     {
         final DataSourceResolver dataSourceResolver = new DataSourceUrlResolver(new URL("http://does.not.exist"), false);
         assertNotNull(dataSourceResolver.resolve("images/asf_logo_wide.gif"));
 
-        dataSourceResolver.resolve("images/does-not-exist.gif");
+        final Executable testMethod = () -> dataSourceResolver.resolve("images/does-not-exist.gif");
+        assertThrows(IOException.class, testMethod);
```

Comment:
Hello @nhojpatrick 
Thank you for your PR.
As with the other PRs, the local variable is superfluous. 
Please rebase on master to fix the spotbugs build on Java 11 and 17.